### ISSUE TITLE
Training slows down with repeated calls to Model.fit()

### ISSUE THREAD

**Issue by dlop** [tensorflow/tensorflow#33315]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 Version 1903
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): v2.0.0-rc2-26-g64c3d382ca 2.0.0
- Python version: 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 17:00:18) [MSC v.1900 64 bit (AMD64)]
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**
Calls to `tf.keras.Model.fit()` run slower and slower the more there have been. The sample code below reports about 200 us/sample at the start but after running for 20 minutes report about 400 us per sample. This continues to increase with no apparent upper limit. It remains slow even when training subsequent models but resets to fast after restarting the program.

**Describe the expected behavior**
No long term upward trend in training time.

**Code to reproduce the issue**

    import numpy
    import tensorflow as tf
    input = tf.keras.Input(shape=(1600,))
    z = tf.keras.layers.Dense(units=200, activation='relu')(input)
    z = tf.keras.layers.Dense(units=1)(z)
    model = tf.keras.Model(inputs=input, outputs=z)
    model.compile(loss='mse')
    x = numpy.full((100,1600), fill_value=2.34, dtype=numpy.float32)
    y = numpy.full((100,1), fill_value=1.23, dtype=numpy.float32)
    while True:
      tf.keras.backend.clear_session()
      model.fit(x=x, y=y)

**Other info / logs**

**Comment by nnnnwang** [comment]
did you solved it?i met the same problem,it really troubles me

**Comment by dlop** [comment]
> did you solved it?i met the same problem,it really troubles me

No. I avoided it by changing from CPU to GPU which was OK.

**Comment by nnnnwang** [comment]
> > did you solved it?i met the same problem,it really troubles me
> 
> No. I avoided it by changing from CPU to GPU which was OK.

sorry,i don't get it. Do you mean that when you train the model using cpu, the training time keeping incresing. But,when you use gpu to train the model , it will be ok?

**Comment by dlop** [comment]
Yes. I only found it slowing down on CPU and see constant speed on GPU. Slowdown can also be caused by overheating, so make sure the CPU clock frequency or GPU load isn't dropping.

**Comment by nnnnwang** [comment]
> Yes. I only found it slowing down on CPU and see constant speed on GPU. Slowdown can also be caused by overheating, so make sure the CPU clock frequency or GPU load isn't dropping.

thx, i solved it

**Comment by saikumarchalla** [comment]
@dlop Please feel free to close the issue if it was resolved already. Thanks!

**Comment by saikumarchalla** [comment]
@dlop  Any updates regarding this issue? Thanks!

**Comment by dlop** [comment]
> @dlop Any updates regarding this issue? Thanks!

I'll try again with TF 2.1.

**Comment by dlop** [comment]
I'm still experiencing this problem with TF 2.0 that I compiled for GPU but set to CPU-only and run on a different problem from the code in this issue. I've confirmed that it's not due to CPU raising temperature or reducing clock speed since those are steady or improving overall. I haven't tried TF 2.1 yet.

Attached file shows the speed halves after about 1 hour of training.

![slowdown](https://user-images.githubusercontent.com/10569080/80658028-29ad9700-8ad9-11ea-8d5f-9845c8632f84.png)
